src/components/ListedSpecies/BandingSection.jsx

```diff
@@ -0,0 +1,286 @@
+import {
+  Box,
+  Button,
+  chakra,
+  Checkbox,
+  Code,
+  FormLabel,
+  Grid,
+  GridItem,
+  HStack,
+  Input,
+  Select as ChakraSelect,
+  Table,
+  TabList,
+  TabPanel,
+  TabPanels,
+  Tabs,
+  Tbody,
+  Td,
+  Text,
+  Thead,
+  Tr,
+} from '@chakra-ui/react';
+import { chakraComponents, Select } from 'chakra-react-select';
+import React, { useState } from 'react';
+import { Controller, useFieldArray, useFormContext } from 'react-hook-form';
+import { IconContext } from 'react-icons';
+import { BsFillCircleFill } from 'react-icons/bs';
+import BandColors from '../../common/BandColors';
+import CollapsibleSection from '../CollapsibleSection/CollapsibleSection';
+import CloseableTab from './CloseableTab';
+
+const BAND_POSITIONS = ['Top Left', 'Top Right', 'Bottom Left', 'Bottom Right'];
+
+const generateSingleBandCode = band => {
+  if (!band.verticalPosition || !band.colors || band.colors.length === 0) return '';
+  let color;
+  if (band.colors.length === 1) {
+    color = band.colors[0].realValue;
+  } else {
+    color = `(${band.colors.map(c => c.realValue).join('/')})`;
+  }
+  if (band.verticalPosition === 'below') {
+    color = color.toLowerCase();
+  }
+  if (band.alphanumeric && band.alphanumeric !== '') {
+    return `${band.flag ? 'F' : ''}${color}.${band.alphanumeric}`;
+  }
+  return `${band.flag ? 'F' : ''}${color}`;
+};
+const generateCode = bands => {
+  if (!bands || !bands[0]) {
+    return 'invalid';
+  }
+  const [topLeft, topRight, bottomLeft, bottomRight] = bands;
+  if (
+    (topLeft.verticalPosition === 'below' && bottomLeft.verticalPosition === 'above') ||
+    (topRight.verticalPosition === 'below' && bottomRight.verticalPosition === 'above')
+  ) {
+    return 'Top band must be above bottom band';
```

Comment:
If it's a must, users should be unable to save to tracker until they fix this?  Maybe same case if invalid

Reply:
added